frontend/deb/distro/install.go

```diff
@@ -0,0 +1,174 @@
+package distro
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend/deb"
+	"github.com/moby/buildkit/client/llb"
+	"github.com/pkg/errors"
+)
+
+// AptInstall returns an [llb.RunOption] that uses apt to install the provided
+// packages.
+//
+// This returns an [llb.RunOption] but it does create some things internally,
+// so any [llb.ConstraintsOpt] you want to pass in should come before this option
```

Comment:
I really don't like the requirement that this has to follow all `llb.ConstraintsOpt`s. It introduces an implicit parameter structure for arguments to a function, that is not enforced in any way. If the caller makes a mistake, the result is a silent and subtle bug.

If we forget to tell a new developer about this, and they miss the comment, we could be in for a bad time (or at least a lost and frustrated developer)